models/models.js

```diff
@@ -23,12 +23,20 @@ exports.selectArticleById = (article_id) => {
     });
 };
 
-exports.selectArticles = () => {
+exports.selectArticles = (topic) => {
+  let queryString = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM comments RIGHT JOIN articles ON articles.article_id = comments.article_id `;
+  const queryValues = [];
+  if (topic) {
+    queryValues.push(topic);
+    queryString += `WHERE topic = $1 `;
+  }
+  queryString += `GROUP BY articles.article_id, articles.created_at ORDER BY articles.created_at DESC;`;
   return db
-    .query(
-      `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM comments RIGHT JOIN articles ON articles.article_id = comments.article_id GROUP BY articles.article_id, articles.created_at ORDER BY articles.created_at DESC;`
-    )
+    .query(queryString, queryValues)
     .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: 'not found' });
+      }
       for (let i = 0; i < rows.length; i++) {
         rows[i].comment_count = +rows[i].comment_count;
       }
```

Comment:
Really nice construction of your query here!

Rather than loping through and changing each of your comment_counts to numbers manually, you may want to look at PSQLs `CAST`, or the `:: INT` syntax. This would be a small change to your query to avoid having to loop.